wordpress-mobile/WordPress-Android
# My Site Dashboard: Tabs - Track experiment property with all events

Parent #16007 

This PR adds support for tracking My Site Default Tab Experiment properties with every event tracked (when the experiment is running)
The following property is tacked on to every event 
  `default_tab_experiment` values =`dashboard|site_menu|nonexistent`

I tried to keep this as generic as possible and not pollute the "tracker" with references to "MySIteDashboard". 

**Merge Instructions**
- Ensure that ~~PR #16175 and #16176 have been merged~~ 
- ~~-  Remove the Not Ready for Merge label~~ 
- Merge as normal

**To test:**
- Launch the app and Login
- If not enabled, please enable `MySiteDashboardTabsFeatureConfig`, then restart the app and logout
- Login
- Navigate to My Site
- View the logs and note that you see property "default_tab_experiment" being tracked on all events
- Navigate to Me -> App Settings -> Debug Settings 
- Disable `my_site_default_tab_experiment` and restart the app
- View the logs and note that you don't see property  "default_tab_experiment" being tracked


## Regression Notes
1. Potential unintended areas of impact
Events are not tracked

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing

3. What automated tests I added (or what prevented me from doing so)
Add unit tests to `MySiteDefaultTabExperimentTest`

PR submission checklist:
